content/codes/components/enigma-path-svg.ts

```diff
@@ -0,0 +1,558 @@
+      (dashoffset ? '" stroke-dashoffset="' + dashoffset : '') +
+      '" fill="' + fill + '"></path>';
+  }
+
+  function pathpts(points: any[], close = false) {
+    return 'M' + points.map(p => p.x + ',' + p.y).join('L') + (close ? 'Z' : '');
+  }
+
+  function addRoundRect(layer:ElementView, x:number, y:number, width:number, height:number, rl:number, rr:number,
+      stroke:string, fill:string) {
+    let path = '<path d="';
+    path += move(x + rl, y);
+    path += line(x + width - rr, y);
+    path += arc(rr, x + width, y + rr);
+    path += line(x + width, y + height - rr);
+    path += arc(rr, x + width - rr, y + height);
+    path += line(x + rl, y + height);
+    path += arc(rl, x, y + height - rl);
+    path += line(x, y + rl);
+    path += arc(rl, x + rl, y);
+    path += '" stroke-width="1" stroke="' + stroke + '" fill="url(#' + fill + ')" />';
+    layer.html += path;
+  }
+
+  const svgdoc:Document = document.getElementById(svgid) as unknown as Document;
+
+  const incolour = 'rgb(205, 14, 102)';
+  const outcolour = 'rgb(34, 171, 36)';
+  const refcolour = 'rgb(15, 130, 242)';
+
+  const reflTopCol = 'rgb(245, 226, 204)';
+  const reflBotCol = 'rgb(178, 160, 143)';
+
+  const rotorTopCol = 'rgb(210, 205, 199)';
+  const rotorBotCol = 'rgb(161, 157, 153)';
+
+  const plugTopCol = 'rgb(204, 226, 245)';
+  const plugBotCol = 'rgb(143, 160, 178)';
+
+  const labelOuterCol = 'rgb(160, 160, 160)';
+  const labelInnerCol = 'rgb(240, 240, 240)';
+
+  const defs = $('.' + svgid + '_defs') as ElementView;
+  const styles = $('.' + svgid + '_styles') as ElementView;
+  const content = $('.' + svgid + '_machine_content') as ElementView;
+  const encpath = $('.' + svgid + '_path_content') as ElementView;
+  const animpath = $('.' + svgid + '_animate_content') as ElementView;
+
+  defs.html += makeLinearGradient('grad_refl', [0, 0, 0, 1],
+      [{offset: 0, col: reflTopCol},
+        {offset: 1, col: reflBotCol}]);
+
+  defs.html += makeLinearGradient('grad_rotor', [0, 0, 0, 1],
+      [{offset: 0, col: rotorTopCol},
+        {offset: 1, col: rotorBotCol}]);
+
+  defs.html += makeLinearGradient('grad_rotor_label', [0, 0, 0, 1],
+      [{offset: 0, col: labelOuterCol},
+        {offset: 0.33, col: labelInnerCol},
+        {offset: 0.67, col: labelOuterCol}]);
+
+  defs.html += makeLinearGradient('grad_plug', [0, 0, 0, 1],
+      [{offset: 0, col: plugTopCol},
+        {offset: 1, col: plugBotCol}]);
+
+  // setup the text styles
+
+  styles.html +=
+    '.heading { font: normal 14px sans-serif; fill:black;}' +
+    '.normal { font: normal 12px sans-serif; fill:black; }' +
+    '.inletter { font: normal 12px sans-serif; fill:' + incolour + '; }' +
+    '.outletter { font: normal 12px sans-serif; fill:' + outcolour + '; }' +
+    '.keyletter { font: normal 12px sans-serif; fill:#D0D0D0; }' +
+    '.inkey { font: normal 12px sans-serif; fill:' + incolour + '; }' +
+    '.outkey { font: normal 12px sans-serif; fill:' + outcolour + '; }';
```

Comment:
Let's add all these styles to the `.less` file in this directory. (inclour and outcolour can still be variables)